tests/yast2_cmd/yast_http_server.pm

```diff
@@ -0,0 +1,109 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+#
+# Summary: Setup http-server and check if it works well.
+# - setup main server
+# - create and setup virtual host
+# - check http server working well
+# - setup other options
+# - stop http server and clean up tmp files
+# Maintainer: Shukui Liu <skliu@suse.com>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use version_utils 'is_sle';
+
+# this is a tmp file for testing
+my $creat_tmp_file = 'cat > /srv/www/htdocs/new_dir/tmp.html << EOF
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="utf-8">
+    <title>hello world</title>
+</head>
+<body>
+hello world!
+</body>
+</html>
+EOF';
+
+sub check_bsc1145399 {
+    my $ret_grep1 = script_run "grep 'SSL 0' /etc/apache2/vhosts.d/localhost.conf ";
+    if ($ret_grep1 == 0) {
+        record_soft_failure "bsc#1145399, yast http-server hosts create ... should not result in syntax error.";
+        assert_script_run "sed -i '/SSL 0/d' /etc/apache2/vhosts.d/localhost.conf";
+        return 1;
+    }
+    return;
+}
+
+sub run {
+    select_console 'root-console';
+    zypper_call 'in yast2-http-server';
+    assert_script_run 'mkdir -p /srv/www/htdocs/new_dir';
+
+    # create a tmp file for testing
+    assert_script_run "$creat_tmp_file\n(exit $?)";
+
+
+    # create and setup virtual host
+    if (is_sle('=15-sp1')) {
+        # setup main server
+        my $output;
+        $output = script_output('yast http-server configure host=main servername=sle serveradmin=root@sle documentroot=/srv/www/htdocs 2>&1', proceed_on_failure => 1);
+        $output =~ m#Internal error# && record_soft_failure "bsc#1145538 yast http-server should not give internal error";
+
+        type_string "yast http-server hosts create servername=localhost serveradmin=admin\@localhost documentroot=/srv/www/htdocs/new_dir \n";
+        wait_still_screen 10;
+        send_key "alt-d";
+        send_key "alt-o";
+        wait_still_screen 10;
+        record_soft_failure "bsc#1145538 yast http-server should not give internal error";
```

Comment:
This `record_soft_failure` now duplicate

Reply:
I think this is unavoidable, because the "internal error" occurs twice: line 62 and 65 both throw it out

Reply:
But you use `record_soft_failure` on line 63, you don't need to run it here again, here is no check for it.

Reply:
now you have two windows with same message ... http://10.67.20.133/tests/939#step/yast_http_server/21

Reply:
maybe I should just check line 63, and remove line 70?

Reply:
the second record_soft_failure is removed.

Reply:
nice, thank you